### PR TITLE
tf2_echo: specifies quaternion order to be xyzw

### LINKED ISSUE #717: tf2_echo: specifying quaternion order (right now ambiguous if xyzw or wxyz)
#### Feature description

When echoing transformation between two frames using `ros2 run tf2_ros tf2_echo foo bar`, the resulting logs containing`Rotation: in Quaternion` does not specify whether the ordering of the 4 elements is `xyzw` or `wxyz`.

Peeking inside the source code, the order is `xyzw`:
https://github.com/ros2/geometry2/blob/ros2/tf2_ros/src/tf2_echo.cpp#L153-L154
```
std::cout << "- Rotation: in Quaternion [" << rotation.x << ", " << rotation.y << ", " << rotation.z << ", " << rotation.w << "]" << std::endl;
```

I propose specifying this order in the output of `tf2_echo`.

#### Implementation considerations

A single `std::cout()` needs to be changed in `tf2_ros/src/tf2_echo.cpp` (linked above).

### PR DESCRIPTION
**Pull request by abhishek47kashyap**
Resolves #717

Targets `rolling` but can be backported to earlier distros that haven't reached EOL (`jazzy`, `iron`, `humble`) because the change is only a single `std::cout`.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
Thank you for the PR, there are some linter issues https://build.ros2.org/job/Rpr__geometry2__ubuntu_noble_amd64/118/

File: tf2_ros/src/tf2_echo.cpp [hunk 1]
```diff
@@ -150,8 +150,8 @@ int main(int argc, char ** argv)
       auto rotation = echo_transform.transform.rotation;
       std::cout << "- Translation: [" << translation.x << ", " << translation.y << ", " <<
         translation.z << "]" << std::endl;
-      std::cout << "- Rotation: in Quaternion [" << rotation.x << ", " << rotation.y << ", " <<
-        rotation.z << ", " << rotation.w << "]" << std::endl;
+      std::cout << "- Rotation: in Quaternion (xyzw) [" << rotation.x << ", " << rotation.y <<
+        ", " << rotation.z << ", " << rotation.w << "]" << std::endl;
```
Comment (abhishek47kashyap) [inline tf2_ros/src/tf2_echo.cpp:154]: Diff `", " <<` is to comply with the linter.

**Comment by ahcorde** [comment]
Pulls: ros2/geometry2#718
Gist: https://gist.githubusercontent.com/ahcorde/5dbc7df10f224c6b7d9e10b5d146297d/raw/f1e3b4899ff9cd9a9c8ca3def77f8de309555361/ros2.repos
BUILD args:   --packages-above-and-dependencies tf2_ros --packages-above-and-dependencies tf2_ros
TEST args:  --packages-above tf2_ros --packages-above tf2_ros
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14735
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22036/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16292/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1554/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22768/)

**Comment by clalancette** [comment]
> Targets `rolling` but can be backported to earlier distros that haven't reached EOL (`jazzy`, `iron`, `humble`) because the change is only a single `std::cout`.

While I'm on board with adding this to `rolling`, I'm hesitant to add it to the stable distributions.  That's because people could be running the existing binary and parsing the output, and this would break that use case.  So I'd say let's get this into Rolling for now, and leave it there.  If you disagree strongly, let me know and I can bring it up with the rest of the ROS PMC to get more input.

**Pull request merged**